feathr_project/feathr/_preprocessing_pyudf_manager.py

```diff
@@ -39,12 +43,12 @@ def build_anchor_preprocessing_metadata(anchor_list, local_workspace_dir):
                 continue
             preprocessing_func = anchor.source.preprocessing
             if preprocessing_func:
-                _PreprocessingPyudfManager.persist_pyspark_udf_to_file(preprocessing_func, local_workspace_dir)
+                _PreprocessingPyudfManager.persist_pyspark_udf_to_file(anchor.source.name, preprocessing_func, local_workspace_dir)
```

Comment:
i use a separate UDF file for ease of management and readability since i am writing the source code.

Since you are using pickle now, you can just put it all in the feature_names_to_func_mapping in one shot.

Reply:
after you combine them, you probably don't need source name any  more. source name is not stable and is less preferred.

Reply:
Here is the thing, if we put pickled code directly inside `feature_names_to_func_mapping`, there will be duplicates if one UDF is used by multiple places, and even worse, if there is any one-shot logics inside of this UDF, it will be executed multiple times, that is semantically different than the original implementation and may even cause bugs.

Reply:
You're right, this part is not necessary, removed.